src/common/latest/sonarqube/Endpoint.ts

```diff
@@ -31,41 +30,38 @@ export default class Endpoint {
     this.data = data;
     // Remove trailing slash at the end of the base url, if any
     if (this.data) {
-      this.data.url = this.data?.url.replace(/\/$/, "");
+      this.data.url = this.data.url.replace(/\/$/, "");
     }
   }
 
   public get url() {
     return this.data.url;
   }
 
-  public get auth(): { username: string; password: string } {
+  public get auth(): { username: string; password?: string } {
     // If using user/password
     if (!this.data.token && this.data.password && this.data.password.length > 0) {
       return { username: this.data.username, password: this.data.password };
     }
     // Using token
-    return { username: this.data.token || this.data.username, password: "" };
```

Comment:
I reverted to what we used before, and it works because it's been validated with token only (which sets password as undefined).

But when checking the types, it looks like we actually should not omit `password`. I'll double check this (why isn't the type check finding this, and what should be used)